geoh5py/objects/surveys/electromagnetics/base.py

```diff
@@ -0,0 +1,415 @@
+#  Copyright (c) 2022 Mira Geoscience Ltd.
+#
+#  This file is part of geoh5py.
+#
+#  geoh5py is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  geoh5py is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import uuid
+from typing import Any
+
+import numpy as np
+
+from geoh5py.data import FloatData
+from geoh5py.groups import PropertyGroup
+from geoh5py.objects import Curve
+from geoh5py.objects.object_type import ObjectType
+
+
+class BaseEMSurvey(Curve):
+    """
+    A base electromagnetics survey object.
+    """
+
+    __METADATA: dict = {"EM Dataset": {}}
+    __INPUT_TYPE = None
+    __UNITS = None
+    _receivers: BaseEMSurvey | None = None
+    _transmitters: BaseEMSurvey | None = None
+
+    def __init__(self, object_type: ObjectType, **kwargs):
+
+        super().__init__(object_type, **kwargs)
+
+    def add_component_data(self, data: dict) -> list[PropertyGroup]:
+        """
+        Add lists of data components to an EM survey. The name of each component is
+        appended to the metadata 'Property groups'.
+
+        Data channels must be provided for every frequency or time
+        in order specified by
+        :attr:`~geoh5py.objects.surveys.electromagnetics.BaseEMSurvey.channels`.
+        The data channels can be supplied as either a list of
+        :obj:`geoh5py.data.float_data.FloatData` entities or :obj:`uuid.UUID`
+
+        .. code-block:: python
+
+            data = {
+                "Component A": [
+                    data_entity_1,
+                    data_entity_2,
+                ],
+                "Component B": [...],
+            },
```

Comment:
**suggestion:** parameters would be less opaque, and the function more self explanatory if accepting a single component 
The function call would be: add_component_data(component_name, data)

Reply:
after second thought: this function can stay, but it would differ to some `add_single_component_data(component_name, data)` in the for loop

Reply:
Broke down the function in two